content/tutorial-reliable-transaction-submission.md

```diff
@@ -104,23 +105,49 @@ On restart, or the determination of a new last validated ledger (pseudocode):
 For each persisted transaction without validated result:
     Query transaction by hash
     If (result appears in validated ledger)
-		Persist the final result
-		If (result code is tesSUCCESS)
-			Application may act based on successful transaction
-		Else
-			Application may act based on failure
-			Maybe resubmit with new LastLedgerSequence and Fee
-	Else if (LastLedgerSequence > newest validated ledger)
-        Wait for more ledgers to be validated
-	Else
-		If (server has contiguous ledger history up to and
-	        including the ledger identified by LastLedgerSequence)
-			The transaction failed
-			Submit a new transaction, if appropriate for application
-		Else
-			Repeat submission of original transaction
+      # Outcome is final
+      Persist the final result
+      If (result code is tesSUCCESS)
+        Application may act based on successful transaction
+      Else
+        The transaction failed (1)
+        If appropriate for the application and failure type, submit with
+            new LastLedgerSequence and Fee
+
+    Else if (LastLedgerSequence > newest validated ledger)
+      # Outcome is not yet final
+      Wait for more ledgers to be validated
+
+    Else
+      If (server has continuous ledger history from the ledger when the
+            transaction was submitted up to and including the ledger identified
+            by LastLedgerSequence)
+        The transaction failed (2)
+        If appropriate for the application, submit with
+            new LastLedgerSequence and Fee
+
+      Else
+        # Outcome is final, but not known due to a ledger gap
+        Wait to acquire continuous ledger history
 ```
 
+#### Failure Cases
+
+The difference between the two transaction failure cases (labeled (1) and (2) in the pseudo-code) is whether the transaction was included in a validated ledger. In both cases, you should decide carefully how to process the failure.
+
+- In failure case (1), the transaction was included in a ledger and destroyed the [XRP transaction cost](concept-transaction-cost.html), but did nothing else. This could be caused by a lack of liquidity, improperly specified [paths](concept-paths.html), or other circumstances. For many such failures, immediately retrying with a similar transaction is likely to have the same result. You may get different results if you wait for circumstances to change.
+
+- In failure case (2), the transaction was not included in a validated ledger, so it did nothing at all, not even destroy the transaction cost. This could be the result of the transaction cost being too low for the current load on the XRP Ledger, the `LastLedgerSequence` being too soon, or simply a coincidence. In contrast to failure case (1), it is more likely that a new transaction is likely to succeed if you change only the `LastLedgerSequence` and possibly the `Fee` and submit again.
```

Comment:
Suggestion:

or simply a coincidence -> or an unreliable server and/or network connection

I guess it could also be due to another transaction, consuming the same account sequence number, getting into the ledger first.

Reply:
There are lots of things that could go wrong, but I agree "simply a coincidence" is too hand-wavey. Changed to:

> or it could be due to other conditions such as an unstable network connection.

I don't want to exclude other things, but that's probably the biggest and most understandable one.